packages/data-stores/src/plans/hooks/use-pricing-meta-for-grid-plans.ts

```diff
@@ -175,7 +184,56 @@ const usePricingMetaForGridPlans = ( {
 				}
 
 				/**
-				 * 2. Original and Discounted prices for plan available for purchase.
+				 * 2. Original and Discounted prices for site-specific plans available for purchase
+				 */
+				if ( siteId && planAvailabilityForPurchase[ planSlug ] ) {
+					const originalPrice = {
+						monthly: getTotalPrice(
+							sitePlan?.pricing.originalPrice.monthly,
+							storageAddOnPriceMonthly
+						),
+						full: getTotalPrice( sitePlan?.pricing.originalPrice.full, storageAddOnPriceYearly ),
+					};
+
+					// Do not return discounted prices if discount is due to plan proration
+					if (
+						! withProratedDiscounts &&
+						sitePlan?.pricing?.costOverrides?.[ 0 ].overrideCode ===
```

Comment:
This throws an error when testing with a fresh free site. There's apparently a plan with `[]` coming from the API. 

<img width="385" alt="Screenshot 2024-07-05 at 1 36 05 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/90bf857d-2197-4844-b135-9ea73aed2f59">

Changing to `sitePlan?.pricing?.costOverrides?.[ 0 ]?.overrideCode` fixed it, but I'm not sure if that's the right solution. I think best if `unpackCostOverrides` below checks for empty values before running the map function and return `null`/`undefined`.

Reply:
I could not reproduce this error. Can you please share testing instructions for this? Irrespective, I'll add the check since the array is not guaranteed to be empty.